pkg/aws/policies.go

```diff
@@ -848,18 +864,40 @@ func (c *awsClient) DeleteInlineRolePolicies(role string) error {
 	return nil
 }
 
-func (c *awsClient) deleteAccountRolePolicies(role *string) error {
-	err := c.detachAttachedRolePolicies(role)
+func (c *awsClient) deleteAccountRolePolicies(role *string) (string, error) {
+	policyMap, err := c.GetPolicies([]string{*role})
 	if err != nil {
-		return err
+		return "", err
+	}
+	err = c.detachAttachedRolePolicies(role)
+	if err != nil {
+		return "", err
 	}
 	err = c.DeleteInlineRolePolicies(aws.StringValue(role))
 	if err != nil {
-		return err
+		return "", err
+	}
+	_, policyARN, err := c.deletePolicies(policyMap[*role])
+	if err != nil {
+		return policyARN, err
 	}
 
-	return nil
+	return "", nil
 }
+
+func (c *awsClient) deletePolicies(policies []string) (*iam.DeletePolicyOutput, string, error) {
+	var output *iam.DeletePolicyOutput
+	var err error
+	for i := range policies {
```

Comment:
NP
```suggestion
	for _, policy := range policies {
```
And then we can remove `policyARN := policies[i]`

Reply:
We cannot do this because we are taking the memory address of `policyARN` -- You cannot take the memory address of loop variables such as `policy` in your suggestion.

Reply:
@oriAdler

Reply:
I have removed the variable policyARN anyway and now am only using one instance of `policies[i]`